server.go

```diff
@@ -262,11 +270,32 @@ func (s *Server) ReadSocket(packetPool *sync.Pool, reuseport bool) {
 
 // HTTPServe starts the HTTP server and listens perpetually until it encounters an unrecoverable error.
 func (s *Server) HTTPServe() {
+	var prf interface {
+		Stop()
+	}
+
+	once := sync.Once{}
+
+	if s.debug {
+		profileStartOnce.Do(func() {
+			prf = profile.Start()
+		})
+
+		defer func() {
+			defer once.Do(prf.Stop)
+		}()
+	}
 	httpSocket := bind.Socket(s.HTTPAddr)
 	graceful.Timeout(10 * time.Second)
 	graceful.PreHook(func() {
+
+		if prf != nil {
+			once.Do(prf.Stop)
+		}
+
 		log.Info("Terminating HTTP listener")
 	})
+	graceful.AddSignal(syscall.SIGUSR2, syscall.SIGHUP)
```

Comment:
nitpick: let's add a comment here that we want to be notified on SIGUSR2 even when not running under einhorn, and that's why we're adding it explicitly. (otherwise, a future reader might think, "well sigusr2 already gets added when you're under einhorn, so adding sigusr2 here is unnecessary")